benchmarks/dbscan/ArborX_DBSCANVerification.hpp

```diff
@@ -46,7 +46,9 @@ bool verifyCorePointsNonnegativeIndex(ExecutionSpace const &exec_space,
         bool self_is_core_point = (offset(i + 1) - offset(i) >= core_min_size);
         if (self_is_core_point && labels(i) < 0)
         {
-#ifdef __SYCL_DEVICE_ONLY__
+#if KOKKOS_VERSION >= 40199
```

Comment:
This way, we can at least test it before the `Kokkos` release and I expect we update all intermediate versions when `Kokkos 4.2.0` is released. Also, this is how we have done it before, see https://github.com/arborx/ArborX/blob/57f78846bdc883bc21be71c2f907465b4a75c4ba/src/kokkos_ext/ArborX_DetailsKokkosExtScopedProfileRegion.hpp#L17-L45 for example. I don't think we need to guarantee that `ArborX` `master` works with all `Kokkos` `develop` commits since the last release.

Reply:
I don't remember the details about scoped profiling region but when we did that with math functions it was merged first thing into develop after we branched off.

Reply:
Ultimately, I'm fine either way. If it helps to move forward here, I'm happy to switch the version check to `4.2.00`.